### PR TITLE
Extract Google Proxy detection from `detect.js`

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Extracts the check for Google’s crawlers outside of `detect.js`.

Similar to #24443.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### CODE REVIEW

File: static/src/javascripts/lib/detect-google-proxy.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * https://developers.google.com/search/docs/advanced/mobile/web-light
+ */
+const GOOGLE_WEB_LIGTH = 'googleweblight';
+
+/**
+ * This one is undocumented, not sure it actually exists.
+ */
+const GOOGLE_WEB_PREVIEW = 'Google Web Preview';
```
Comment (mxdvl) [inline static/src/javascripts/lib/detect-google-proxy.ts:9]: This is very odd…. I don’t know why it was added in the first place.

**Review by arelra — APPROVED** [review]

File: static/src/javascripts/lib/detect-google-proxy.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * https://developers.google.com/search/docs/advanced/mobile/web-light
+ */
+const GOOGLE_WEB_LIGTH = 'googleweblight';
```
[the remark this answers is not in the document]
Comment (arelra) [reply static/src/javascripts/lib/detect-google-proxy.ts:4]:
```suggestion
const GOOGLE_WEB_LIGHT = 'googleweblight';
```

File: static/src/javascripts/lib/detect-google-proxy.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+const GOOGLE_WEB_PREVIEW = 'Google Web Preview';
+
+/**
+ * Read more about Google Crawler here: https://developers.google.com/search/docs/advanced/crawling/overview-google-crawlers
+ * @returns whether this is a Google Proxy
+ */
+export const isGoogleProxy = (): boolean =>
+	Boolean(
+		navigator.userAgent.includes(GOOGLE_WEB_PREVIEW) ||
+			navigator.userAgent.includes(GOOGLE_WEB_LIGTH),
```
Comment (arelra) [inline static/src/javascripts/lib/detect-google-proxy.ts:18]:
```suggestion
			navigator.userAgent.includes(GOOGLE_WEB_LIGHT),
```

File: static/src/javascripts/lib/detect-google-proxy.ts (hunk 2 above)
Comment (arelra) [inline static/src/javascripts/lib/detect-google-proxy.ts:4]:
```suggestion
const GOOGLE_WEB_LIGHT = 'googleweblight';
```

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.ts [hunk 4]
```diff
@@ -5,21 +5,23 @@ import {
...
 import config from '../../../../lib/config';
-import { isGoogleProxy } from '../../../../lib/detect';
 import { getPageTargeting } from '../../../common/modules/commercial/build-page-targeting';
 import { commercialFeatures } from '../../../common/modules/commercial/commercial-features';
 import { prebid } from '../header-bidding/prebid/prebid';
 import { shouldIncludeOnlyA9 } from '../header-bidding/utils';
 import { dfpEnv } from './dfp-env';
 
 const loadPrebid = async (framework: Framework): Promise<void> => {
+	// TODO: Understand why we want to skip A9 for Google Proxy
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.ts:17]:
```suggestion
	// TODO: Understand why we want to skip Prebid for Google Proxy
```

**Pull request merged**